afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/WebSocketEventsBus.java

```diff
@@ -29,25 +24,18 @@
 
     private final WeakListenerList<AppStorageListener> listeners = new WeakListenerList<>();
 
-    private NodeEventClient nodeEventClient;
+    private final NodeEventClient nodeEventClient;
 
-    private AppStorage storage;
+    private final AppStorage storage;
 
-    public WebSocketEventsBus(AppStorage storage, URI restUri) {
+    public WebSocketEventsBus(AppStorage storage, URI restUri, WebsocketConnectionPolicy connectionPolicy) {
         URI wsUri = SocketsUtils.getWebSocketUri(restUri);
         this.storage = Objects.requireNonNull(storage);
         URI endPointUri = URI.create(wsUri + "/messages/" + AfsRestApi.RESOURCE_ROOT + "/" +
                 AfsRestApi.VERSION + "/node_events/" + storage.getFileSystemName());
 
-        WebSocketContainer container = ContainerProvider.getWebSocketContainer();
-        try {
-            nodeEventClient = new NodeEventClient(storage.getFileSystemName(), listeners);
-            container.connectToServer(nodeEventClient, endPointUri);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        } catch (DeploymentException e) {
-            throw new UncheckedDeploymentException(e);
-        }
+        nodeEventClient = new NodeEventClient(connectionPolicy.newConnectionManager(endPointUri), storage.getFileSystemName(), listeners);
+        nodeEventClient.connect();
```

Comment:
Maybe the connexion could be done in the constructor instead of calling a `connect` method. From my point of view, it would be more logic with an autocloseable obsject

Reply:
Indeed I wanted to do this, but it did not seem very clean to me:
it will require `webSocketContainer.connectToServer(this, url)` to be called before the constructor of `this` has finished its execution.
Not sure it could be an issue but just in case ... do you have an opinion on this ?

Reply:
(and the connection may call the `onOpen` callback before the end of the constructor, which may not be an issue, but again does not seem very clean to me)

Reply:
I agree to keep the `connect()` method, but then what it's not totally clean is that the `disconnect` method doesn't exist. It's replaced by the close method which is not really symmetrical.

I suppose we can keep the code like this, but in general `AutoCloseable` objects get the ressources in the constructor (see InputStream, OuputStream...), and classes that have a `connect()` method implement `Closeable` (see Socket).

Reply:
Ok, I leave it as is, I agree it's not perfect though.